src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -198,6 +208,54 @@ public sealed class DiagnosticLogManager : AgentService, IDiagnosticLogManager
             executionContext.Debug("Diagnostic file upload complete.");
         }
 
+        // Dumping Agent Azure VM extension logs to the support files folder.
+        private void DumpAgentExtensionLogs(IExecutionContext executionContext, string supportFilesFolder, DateTime jobStartTimeUtc)
+        {
+            executionContext.Debug("Starting dumping agent extension logs.");
+
+            string pathToLogs = null;
+            string archiveName = null;
+            string timestamp = jobStartTimeUtc.ToString("yyyyMMdd-HHmmss");
+
+            if (PlatformUtil.RunningOnWindows)
+            {
+                string pathToVersions = "C:\\WindowsAzure\\Logs\\Plugins\\Microsoft.VisualStudio.Services.TeamServicesAgent";
+                if (Directory.Exists(pathToVersions))
```

Comment:
Could you please add a debug message if the target folder with logs will not be found on Azure VM?